assignments/lambda-classes.js

```diff
@@ -1 +1,112 @@
 // CODE here for your Lambda Classes
+
+class Person { 
+    constructor(baseclass) {
+        this.name = baseclass.name,
+        this.age = baseclass.age,
+        this.location = baseclass.location;
+    }
+    speak () {
+        return `Hello my name is ${this.name}, I am from ${this.location}.`;
+    }
+}
+
+class Instructor extends Person {
+    constructor(teacher) {
+        super(teacher);
+        this.speciality = teacher.speciality,
+        this.favLanguage = teacher.favLanguage,
+        this.catchPhrase = teacher.catchPhrase;
+    }
+    demo(subject) {
+        return `Today we are learning about ${subject}`;
+    }
+    grade(student, subject) {
+       return `${student.name} receives a perfect score on ${subject}`;
+    }
+}
+
+class Student extends Person {
+    constructor(scholar) { 
+        super(scholar);
+        this.previousBackground = scholar.previousBackground,
+        this.className = scholar.className,
+        this.favSubjects = scholar.favSubjects,
+        this.grade = scholar.grade; //stretch challenge, trying to figure this out
+    }
+    listsSubjects () { //Im not sure if this is the correct way of doing this.
+       return `${this.favSubjects}`;
+    }
+    PRAssignment (student, subject) {
+      return `${student.name} has submitted a PR for ${subject}`;
+    }
+    sprintChallenge (student, subject) {
+      return `${student.name} has begun sprint challenge on ${subject}`;
+    }
+    grade () 
```

Comment:
If i'm not mistaken you're wanting this to be the graduate method and not grade according to the stretch goals.

an example for a graduate method could be:
`graduate(){
    if(this.grade >= 70){
        return `Congrats ${this.name} you have graduated!`;
    } else {
        return `${this.name} keep trying you're almost there!`;
    }
}`

this is referring to this particular student object that is calling the graduate method